mlflow/projects/_project_spec.py

```diff
@@ -154,7 +155,12 @@ def compute_parameters(self, user_parameters, storage_dir):
 
         for key, param_obj in self.parameters.items():
             value = user_parameters[key] if key in user_parameters else self.parameters[key].default
-            final_params[key] = param_obj.compute_value(value, storage_dir)
+            target_sub_dir = 'param_{}'.format(list(self.parameters.keys()).index(key))
```

Comment:
@trangevi I've confirmed that the order of a dictionary's key set should be stable across calls to `keys()`, as long as the dictionary is not modified. That said, perhaps it's ever-so-slightly more efficient to fetch the key set and convert it to a list before entering the loop? This would also make it more obvious that the set of values we're using for indexing cannot change across loop iterations (I had to do some StackOverflow-ing to confirm this with the current approach - https://stackoverflow.com/questions/835092/python-dictionary-are-keys-and-values-always-the-same-order).